packages/adapters/eth/src/websocket.rs

```diff
@@ -1,430 +1,464 @@
-use std::{num::NonZeroU32, ops::RangeInclusive, str::FromStr, time::Duration};
+use std::{cmp::min, num::NonZeroU32, ops::RangeInclusive};
 
-use ::metrics::{HealthChecker, RegistersMetrics, prometheus::core::Collector};
 use alloy::{
-    consensus::SignableTransaction,
-    network::TxSigner,
-    primitives::{Address, B256, ChainId},
-    rpc::types::FeeHistory,
-    signers::{Signature, local::PrivateKeySigner},
+    consensus::Transaction,
+    eips::{BlockNumberOrTag, eip4844::DATA_GAS_PER_BLOB},
+    network::{Ethereum, EthereumWallet, TransactionBuilder, TransactionBuilder4844, TxSigner},
+    primitives::Address,
+    providers::{
+        Provider, ProviderBuilder, SendableTx, WsConnect,
+        utils::{EIP1559_FEE_ESTIMATION_PAST_BLOCKS, Eip1559Estimation},
+    },
+    pubsub::PubSubFrontend,
+    rpc::types::{FeeHistory, TransactionReceipt, TransactionRequest},
+    sol,
 };
-use delegate::delegate;
-use serde::Deserialize;
+use itertools::Itertools;
 use services::{
-    Result,
     state_committer::port::l1::Priority,
     types::{
-        BlockSubmissionTx, Fragment, FragmentsSubmitted, L1Height, L1Tx, NonEmpty,
-        TransactionResponse, U256,
+        BlockSubmissionTx, Fragment, FragmentsSubmitted, L1Tx, NonEmpty, TransactionResponse, U256,
     },
 };
+use tracing::info;
 use url::Url;
 
-use self::{
-    connection::WsConnection,
-    health_tracking_middleware::{EthApi, HealthTrackingMiddleware},
+use crate::{
+    Error, Result, blob_encoder,
+    estimation::{MaxTxFeesPerGas, TransactionRequestExt},
+    provider::L1Provider,
+    websocket::metrics::Metrics,
 };
-use crate::{AwsClient, AwsConfig, fee_api_helpers::batch_requests};
 
-mod connection;
-mod health_tracking_middleware;
+pub mod config;
+pub mod factory;
+mod metrics;
+
+pub type WsProvider = alloy::providers::fillers::FillProvider<
+    alloy::providers::fillers::JoinFill<
+        alloy::providers::fillers::JoinFill<
+            alloy::providers::Identity,
+            alloy::providers::fillers::JoinFill<
+                alloy::providers::fillers::GasFiller,
+                alloy::providers::fillers::JoinFill<
+                    alloy::providers::fillers::BlobGasFiller,
+                    alloy::providers::fillers::JoinFill<
+                        alloy::providers::fillers::NonceFiller,
+                        alloy::providers::fillers::ChainIdFiller,
+                    >,
+                >,
+            >,
+        >,
+        alloy::providers::fillers::WalletFiller<EthereumWallet>,
+    >,
+    alloy::providers::RootProvider<alloy::pubsub::PubSubFrontend>,
+    alloy::pubsub::PubSubFrontend,
+    Ethereum,
+>;
```

Comment:
Ugh... This will render horribly in error messages and any place that isn't aware of the type alias. Perhaps we should define a newtype over it to make it more maintainable? Or any other ideas on how to manage this behemoth of a type signature?